### ISSUE TITLE
The Gradle plugin generates invalid Dockerfile when choosing Java 21

### ISSUE THREAD

**Issue by saw303** [micronaut-projects/micronaut-gradle-plugin#923]
### Expected Behavior

If the Gradle build set Java version 21 like

```kotlin
java {
    sourceCompatibility = JavaVersion.toVersion("21")
    targetCompatibility = JavaVersion.toVersion("21")
}
```

the plugin should use the compatible base images for `Dockerfile` and `DockerfileNative`

### Description

When creating a Micronaut application that uses Java 21 

```sh
mn create-app --build=gradle_kotlin --jdk=21 --lang=java --test=junit com.example.demo
```

and the run `./gradlew dockerfile` or `./gradlew dockerfileNative`, Micronaut generates Dockerfile based on a Java 17 runtime. The native image cannot be built and the JAR based Docker image fails to start.

#### DockerfileNative

```Dockerfile
FROM ghcr.io/graalvm/native-image-community:17-ol9 AS graalvm
...
```
#### Dockerfile

```Dockerfile
FROM eclipse-temurin:17-jre-focal
...
```

#### Workaround

As a workaround for the JAR based Dockerfile I can define a different base image such as

```kotlin
asks.named<MicronautDockerfile>("dockerfile") {
    baseImage = "azul/zulu-openjdk:21-jre"
}
```

For the DockerfileNative it seems there is no way to change the GraalVM image, or am I missing something?

### Steps To Reproduce

see actual behaviour...

### Version

4.2.3

**Comment by saw303** [comment]
relates to https://github.com/micronaut-projects/micronaut-core/issues/10046

**Comment by melix** [comment]
This is actually more a problem of Micronaut Starter which should generate the proper code for the docker file tasks (which have the `jdkVersion` property). Part of the problem with doing that automatically (or inferring, if you prefer) is that there is no standard way in Gradle to get that information consistently (we'd have to figure out if `sourceCompatibility` is used, or toolchains, or both).

**Comment by saw303** [comment]
@melix yes, I was not sure whether this is a starter or a Gradle plugin issue. Feel free to move this issue to the right repo.
